jfr-streaming/src/test/java/io/opentelemetry/contrib/jfr/metrics/JfrThreadCountTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jfr.metrics;
+
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.DAEMON;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.UNIT_THREADS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.SumData;
+import org.junit.jupiter.api.Test;
+
+class JfrThreadCountTest extends AbstractMetricsTest {
+  private static final int SAMPLING_INTERVAL = 1000;
+
+  private static void doWork() throws InterruptedException {
+    Thread.sleep(2 * SAMPLING_INTERVAL);
+  }
+
+  private static boolean isDaemon(LongPointData p) {
+    return p.getAttributes().asMap().get(AttributeKey.booleanKey(DAEMON)).equals(true);
```

Comment:
There are actually 3 possible states for the `DAEMON` attribute: true, false and null. We're not expecting a null value since it's invalid in this context, so let's add an assertion for that:
```suggestion
    Boolean daemon = p.getAttributes().get(AttributeKey.booleanKey(DAEMON));
    assertThat(daemon).isNotNull();
    return daemon;
```